wgpu-hal/src/gles/queue.rs

```diff
@@ -214,21 +217,58 @@ impl super::Queue {
             }
             C::CopyBufferToBuffer {
                 src,
-                src_target,
+                src_target: _,
                 dst,
                 dst_target,
                 copy,
             } => {
-                gl.bind_buffer(src_target, Some(src));
-                gl.bind_buffer(dst_target, Some(dst));
+                let is_index_buffer_only_element_dst = !self
+                    .shared
+                    .private_caps
+                    .contains(super::PrivateCapabilities::INDEX_BUFFER_ROLE_CHANGE)
+                    && dst_target == glow::ELEMENT_ARRAY_BUFFER;
 
-                gl.copy_buffer_sub_data(
-                    src_target,
-                    dst_target,
-                    copy.src_offset as i32,
-                    copy.dst_offset as i32,
-                    copy.size.get() as i32,
-                );
+                let copy_src_target = glow::COPY_READ_BUFFER;
```

Comment:
what if we are copying *from* an index buffer. I think `COPY_READ_BUFFER` would not be allowed there too